### PR TITLE
[#12048] Add SQL email generator unit test

### PR DESCRIPTION
**Pull request by NicolasCwy**
Part of #12048 

**Outline of Solution**
- Migrate the SQL email generator `testGenerateSessionLinksRecoveryEmail`
- Migrate test data for this test from `EmailGenerator.json` to `SqlEmailGenerator.json`
- Amend one test data in `EmailGenerator.json` which was erroneous

**FYI**
Other tests in SQL email generator will be migrated in a separate PR, merging this before I make changes to the SQL email generator in PR #12712

### CODE REVIEW

**Comment by NicolasCwy** [comment]
This PR is blocked for now as I can't figure out why the SqlEmailGeneratorTest is affecting  causing the error when component tests are run `teammates.storage.api.FeedbackResponseCommentsDbTest.tearDownLocalDatastoreHelper`

I have a feeling its because my test is `BaseTestCaseWithSqlDatabaseAccess` and I am using a data bundle that is postgres data it somehow is affecting it, will look into this after I finish work in #12712.

**Comment by NicolasCwy** [comment]
Initially I put the `EmailGenerator` test for the SQL email generator into `src/test/java/teammates/sqllogic/api/` to mimic the data store email generator found in `src/test/java/teammates/logic/api/MockUserProvision.java` but got a seemingly unrelated error when component testing. I also realised that the tests adjacent to the above SQL folder did not make use of `BaseTestCaseWithSqlDatabaseAccess` class which the email generator test used. 

I moved my test to `src/it/java/teammates/it/sqllogic/api/` where other tests that used `BaseTestCaseWithSqlDatabaseAccess` were found and it worked. Would this change be ok from a file structure point of view? I'm also not too sure why this fixed it, if you have any insight ont his @cedricongjh 

image

**Comment by cedricongjh** [comment]
> Initially I put the `EmailGenerator` test for the SQL email generator into `src/test/java/teammates/sqllogic/api/` to mimic the data store email generator found in `src/test/java/teammates/logic/api/MockUserProvision.java` but got a seemingly unrelated error when component testing. I also realised that the tests adjacent to the above SQL folder did not make use of `BaseTestCaseWithSqlDatabaseAccess` class which the email generator test used.
> 
> I moved my test to `src/it/java/teammates/it/sqllogic/api/` where other tests that used `BaseTestCaseWithSqlDatabaseAccess` were found and it worked. Would this change be ok from a file structure point of view? I'm also not too sure why this fixed it, if you have any insight ont his @cedricongjh
> 
> image

The tests in the package `src/test/java/teammates/sqllogic/api/` are meant for unit tests, without database access (the package was likely not setup to connect to the db, hence the errors), while the tests in `src/it/java/teammates/it` are meant for integration tests, with access to the database, hence from a file structure point of view this is fine.

File: src/it/java/teammates/it/sqllogic/api/EmailGeneratorTestIT.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        FeedbackSession session1InCourse4 = dataBundle.feedbackSessions.get("session1InCourse4");
+        FeedbackSession session2InCourse4 = dataBundle.feedbackSessions.get("session2InCourse4");
+        // opened and unpublished.
+        session1InCourse3.setStartTime(TimeHelper.getInstantDaysOffsetFromNow(-20));
+        dataBundle.feedbackSessions.put("session1InCourse3", session1InCourse3);
+
+        // closed and unpublished
+        session2InCourse3.setStartTime(TimeHelper.getInstantDaysOffsetFromNow(-19));
+        session2InCourse3.setEndTime(TimeHelper.getInstantDaysOffsetFromNow(-1));
+        // session2InCourse3.setDeletedTime(null);
```
Comment (cedricongjh) [inline src/it/java/teammates/it/sqllogic/api/EmailGeneratorTestIT.java:51]: nit: remove this comment if not needed

**Review by cedricongjh — APPROVED** [review]
LGTM! other than the nit

File: src/it/java/teammates/it/sqllogic/api/EmailGeneratorTestIT.java (hunk 1 above)
Comment (NicolasCwy) [reply src/it/java/teammates/it/sqllogic/api/EmailGeneratorTestIT.java]: Resolved should I get another reviewer or should this be merged?

**Pull request merged**